chains/solana/contracts/programs/ccip-router/src/instructions/v1/fee_quoter.rs

```diff
@@ -74,27 +74,48 @@ pub fn fee_for_msg(
             .unwrap_or_else(|| dest_chain.config.default_tx_gas_limit.into()),
     );
 
+    let calldata_length =
+        U256::new(message.data.len() as u128) + network_fee.transfer_bytes_overhead;
+
+    // Calculate calldata gas cost while accounting for EIP-7623 variable calldata gas pricing
+    // This logic works for EVMs post Pectra upgrade, while being backwards compatible with pre-Pectra EVMs.
+    let calldata_gas = if calldata_length
+        > U256::new(dest_chain.config.dest_gas_per_payload_byte_threshold as u128)
+    {
+        U256::new(dest_chain.config.dest_gas_per_payload_byte_base as u128)
+            * U256::new(dest_chain.config.dest_gas_per_payload_byte_threshold as u128)
+            + (calldata_length
+                - U256::new(dest_chain.config.dest_gas_per_payload_byte_threshold as u128))
+                * U256::new(dest_chain.config.dest_gas_per_payload_byte_high as u128)
```

Comment:
Nit: I know this follows the reference EVM implementation verbatim, but I do wonder if it would be possible to find meaningful names for these individual calculations, so we can extract them into variables.

Currently this nested calculation is a bit difficult to parse visually, so it would be good to give each factor a variable.

Reply:
+1, it looks like it matches EVM but I also find it somewhat hard to read as-is